### DISCUSSION TITLE
use magick to re-encode EXR without a specific compression feature?

### DISCUSSION THREAD

**Discussion by jvenge** [ImageMagick/ImageMagick#6886] [Help]
### ImageMagick version

6.q16

### Operating system, version and so on

Ubuntu 20.04LTS on WSL2

### Description

Hi, I need to re-compress a batch of EXR files that have been saved with the Photoshop EXR plugin with "Luminance/Chroma compression" enabled. Now is supposed to be used in NV Iray, which recently dropped support when switching libraries. 
I used convert before but never on EXR and am lost in the options. Where can I read up on EXR specific things, and can IM do what I seek to begin with?

**Comment by urban-warrior** [comment]
ImageMagick supports EXR images, but its integration is optional. To check if EXR support is enabled, type `convert --version` and see if `openexr` is listed as a delegate. If it is, you can utilize ImageMagick to both read and write EXR images. ImageMagick offers the capability to write RGBA, Y, YC, and YCA images. For instance, you can specify the color type with `-define exr:color-type=YCA`. Additionally, you have the flexibility to define the sampling factor using the `-sampling-factor` option. It's important to note that pixels are written as 16-bit floats.

**Comment by jvenge** [comment]
openexr is a delegate indeed but still I'd need an overview of exr specific switches. I guess in the src, but are there docs?

**Reply by fmw42** [reply]
The place to look should be for EXR at the -defines page. See https://imagemagick.org/script/defines.php. However, there is only one define there for openexr format. See also EXR at the formats page at https://imagemagick.org/script/formats.php

**Comment by snibgo** [comment]
Writing EXR is also sensitive to the `-compress` setting.
